src/ripple/protocol/impl/UintTypes.cpp

```diff
@@ -45,59 +59,50 @@ std::string to_string(Currency const& currency)
 
     if ((currency & sIsoBits).isZero ())
     {
-        // The offset of the 3 character ISO code in the currency descriptor
-        int const isoOffset = 12;
-
         std::string const iso(
-            currency.data () + isoOffset,
-            currency.data () + isoOffset + 3);
+            currency.data () + detail::isoCodeOffset,
+            currency.data () + detail::isoCodeOffset + detail::isoCodeLength);
 
         // Specifying the system currency code using ISO-style representation
         // is not allowed.
         if ((iso != systemCurrencyCode()) &&
-            (iso.find_first_not_of (allowed_characters) == std::string::npos))
+            (iso.find_first_not_of (detail::isoCharSet) == std::string::npos))
         {
             return iso;
         }
     }
 
-    return strHex (currency);
+    return strHex(currency);
 }
 
 bool to_currency(Currency& currency, std::string const& code)
 {
-    if (code.empty () || !code.compare (systemCurrencyCode()))
+    if (code.empty () || !code.compare(systemCurrencyCode()))
     {
         currency = beast::zero;
         return true;
     }
 
-    static const int CURRENCY_CODE_LENGTH = 3;
-    if (code.size () == CURRENCY_CODE_LENGTH)
+    // Handle ISO-4217-like 3-digit character codes.
+    if (code.size () == detail::isoCodeLength)
     {
-        Blob codeBlob (CURRENCY_CODE_LENGTH);
+        if (code.find_first_not_of(detail::isoCharSet) != std::string::npos)
+            return false;
```

Comment:
Making a note that I need to look at this change more closely. It looks like the new code does extra error checking that the old code does not and I want to make sure we don't break anything with the new check.

Reply:
I looked and I don't see how the extra error checking could be transaction breaking. So resolving this comment.